tests/D2L.CodeStyle.Analyzers.Test/Specs/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+﻿// analyzer: D2L.CodeStyle.Analyzers.Language.RequireNamedArgumentsAnalyzer
+
+namespace D2L {
+	public static class Foo {
+		public void _arg0() { }
+		public void _arg1( int a1 ) { }
+		public void _arg2( int a1, int a2 ) { }
+		public void _arg3( int a1, int a2, int a3 ) { }
+		public void _arg4( int a1, int a2, int a3, int a4 ) { }
+		public void _arg5( int a1, int a2, int a3, int a4, int a5 ) { }
+		public void _arg6( int a1, int a2, int a3, int a4, int a5, int a6 ) { }
+		public void _arg7( int a1, int a2, int a3, int a4, int a5, int a6, int a7 ) { }
+		public void _arg8( int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8 ) { }
+		public void _arg9( int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8, int a9 ) { }
+		public void _arg10( int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8, int a9, int a10 ) { }
+		public void _arg11( int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8, int a9, int a10, int a11 ) { }
+
+		public void funcWithParams( int a, int b, int c, params int[] ps ) { }
+
+		public delegate void delegate0Args();
+		public delegate void delegate1Args( int a1 );
+		public delegate void delegate10Args( int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8, int a9, int a10 );
+
+		public static void Test() {
+			#region "low" number of args doesn't require naming
+			_arg0();
+			_arg1( 1 );
+			_arg2( 1, 2 );
+			_arg3( 1, 2, 3 );
+			_arg4( 1, 2, 3, 4 );
+			_arg5( 1, 2, 3, 4, 5 );
+			_arg6( 1, 2, 3, 4, 5, 6 );
+			_arg7( 1, 2, 3, 4, 5, 6, 7 );
+			_arg8( 1, 2, 3, 4, 5, 6, 7, 8 );
+			_arg9( 1, 2, 3, 4, 5, 6, 7, 8, 9 );
+			#endregion
+
+			#region diagnostic for too many unnamed args
+			/* UseNamedArgsForInvocationWithLotsOfArgs */ _arg10( 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ) /**/;
+			/* UseNamedArgsForInvocationWithLotsOfArgs */ _arg11( 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ) /**/;
+			#endregion
+
+			#region all named args is usually preferred if there are lots of args
+			_arg10(
+				a1: 1,
+				a2: 2,
+				a3: 3,
+				a4: 4,
+				a5: 5,
+				a6: 6,
+				a7: 7,
+				a8: 8,
+				a9: 9,
+				a10: 10
+			);
+			#endregion
+
+			#region named args don't count against the unnamed args budget

```

Comment:
I find this questionable, but cool to start, and maybe reasonable as the limit gets dropped

Reply:
Yeah, I am of two minds. I sometimes do it like [here](https://github.com/j3parker/D2L.CodeStyle/blob/a486ce8d7162a23f21fed6ad09d50e0eea2981c0/src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs#L79) when the first args are obvious but then there are defaulted arguments, etc.

Not super attached to it.